### DISCUSSION TITLE
CpModel Copy/Clone in C#

### DISCUSSION THREAD

**Discussion by codecontemplator** [google/or-tools#3000] [CP-SAT questions]
Language: C#
Package: SAT solver

I have a model and I want to explore the feasibility of different sets of constraints. I would like to clone the model to create a snapshot and revert back to that snapshot when the model becomes infeasible to try other versions.

In Python and C++ a **CpModel.CopyFrom** method exists (judging from the docs at https://github.com/google/or-tools/blob/stable/ortools/sat/doc/model.md) and that seems to do what I want.

**Comment by lperron** [comment]
Duly noted. I do not have the time to work on that. So not in the next release. But hopefully soon.
